### DISCUSSION TITLE
Would be a cool feature if you can create masks for full body too, not just the face.

### DISCUSSION THREAD

**Discussion by sahildeshpande** [kex0/batch-face-swap#53] [Ideas]
Because face swap could lead to mismatch in color and tone compared to the full body, it could be useful to also provide full body swaps.

**Comment by kex0** [comment]
I was experimenting with body swapping in this branch https://github.com/kex0/batch-face-swap/tree/body_mask
But I abandoned it because if I remember correctly, it wasn't giving me very good results. Especially on the photos with high resolution.

The reason why face inpainting works well is because it usually only focuses on the small part of the image.
Since bodies usually occupy most of the screen it means that it will inpaint entire image which means it it has to focus on the entire image so it won't give results with higher fidelity than the original image

**Reply by sahildeshpande** [reply]
Interesting. But I think combined with inpainting or the image splitting feature (that already exists) it could be easier? Lets say i inpaint a section of the image that contains the body, so then it searches for the body in that part only

**Reply by kex0** [reply]
The way the `Inpaint` / `Inpaint upload` / `Batch Face Swap` works is: 
1. You create or otherwise provide a mask
2. It crops a rectangle image that envelops your mask.
    <details>
        <summary>Click me!</summary>
        In this example imagine we are inpainting the face. I marked the crop rectangle with red.
        <img src="https://user-images.githubusercontent.com/46696708/236653716-957131b5-f2ea-4618-9f73-317403c9ba57.png">
    </details>
3. It does regular img2img on the cropped image.
4. Finally, it composites the new image to the original image using the mask you provided.
 

So if you provide a mask that is this large, the crop rectangle will basically envelop the entire picture.
<details>
        <summary>Click me!</summary>
        In this example imagine we are inpainting the entire body. I marked the crop rectangle with red.
        <img src="https://user-images.githubusercontent.com/46696708/236653857-5c4bd9e1-6e6d-4c1a-aa92-f8fd1009fdcc.png">
</details>

If we were to split the image and do the img2img part in smaller chunks we could achieve a higher fidelity but body parts would not match when stitched back together.

**Reply by sahildeshpande** [reply]
Ah i see. Makes sense. This requirement will probably need a different technology and logic altogether.
